### PR TITLE
Release `Version 3.0.1`

### PR DESCRIPTION
**Pull request by HCastano**
Follow up to #1189.

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
## 🦑 📈 ink! Example Contracts ‒ Changes Report 📉 🦑
 ⚠️ **The ink! `master` is ahead of your branch, this might skew the comparison data below.**

These are the results when building the `examples/*` contracts from this branch with `cargo-contract 0.18.0-9362c04` and comparing them to ink! `master`: 

|                          | Δ Optimized Size | Δ Used Gas | Total Optimized Size | Total Used Gas |
|:--------------------------|------------------:|------------:|----------------------:|----------------:|
| `accumulator`            |                  |            | 1.03 K               |                |
| `adder`                  |                  |            | 2.17 K               |                |
| `contract-introspection` |                  |            | 2.37 K               |                |
| `contract-terminate`     |                  |            | 0.94 K               | 214_418        |
| `contract-transfer`      |                  |            | 8.15 K               | 14_418         |
| `delegator`              |                  |            | 6.37 K               | 46_295         |
| `dns`                    |                  |            | 8.84 K               | 43_254         |
| `erc1155`                |                  |            | 17.23 K              | 86_508         |
| `erc20`                  |                  |            | 8.49 K               | 43_254         |
| `erc721`                 |                  |            | 11.81 K              | 115_344        |
| `flipper`                |                  |            | 1.31 K               | 14_418         |
| `incrementer`            |                  |            | 1.21 K               | 14_418         |
| `multisig`               |                  |            | 25.20 K              | 93_527         |
| `proxy`                  |                  |            | 2.90 K               | 29_536         |
| `rand-extension`         |                  |            | 3.92 K               | 14_418         |
| `subber`                 |                  |            | 2.19 K               |                |
| `trait-erc20`            |                  |            | 8.76 K               | 43_254         |
| `trait-flipper`          |                  |            | 1.00 K               | 14_418         |
| `trait-incrementer`      |                  |            | 1.19 K               | 28_836         |
| `upgradeable-contract`   |                  |            | 2.99 K               | 14_950         |
| `upgradeable-flipper`    |                  |            | 1.55 K               |                |



[Link to the run](https://gitlab.parity.io/parity/ink-waterfall/-/pipelines/184218) | Last update: Thu Mar 17 18:31:25 CET 2022

**Comment by cmichi** [comment]
As a follow-up to https://github.com/paritytech/ink/pull/1200 you should bump the versions of all crates and update the `RELEASES.md` as part of this PR. You can also be the first one to try out the checklist by releasing 3.0.1.

**Comment by HCastano** [comment]
> As a follow-up to #1200 you should bump the versions of all crates and update the `RELEASES.md` as part of this PR. You can also be the first one to try out the checklist by releasing 3.0.1.

Will try the new release checklist tomorrow 😃

**Comment by cmichi** [comment]
Don't forget to bump the version in all crates + examples.

File: RELEASES.md [hunk 1]
```diff
@@ -1,3 +1,14 @@
+# [Unreleased]
+
+# Version 3.0.1
+
+## Changed
+- Improve upgradeable examples folder structure, explain differences [#1188](https://github.com/paritytech/ink/pull/1188)
+
+## Fixed
+- Update codegen after SCALE v3.1.2 release [#1189](https://github.com/paritytech/ink/pull/1189)
+- Stop using CallData in Multisig doc test [#1202](https://github.com/paritytech/ink/pull/1202)
```
Comment (cmichi) [inline RELEASES.md:10]:
```suggestion
- Update codegen after SCALE v3.1.2 release ‒ [#1189](https://github.com/paritytech/ink/pull/1189)
- Stop using CallData in Multisig doc test ‒ [#1202](https://github.com/paritytech/ink/pull/1202)
```

File: RELEASES_CHECKLIST.md [hunk 2]
```diff
@@ -21,26 +18,50 @@ Reasons for doing it this way:
...
 * Third party tooling like dependabot can easily extract the changelog.
 
+## Examples
+
+Examples (so anything in the `examples/` folder) are a bit of a special case in our
+release pipeline since they are considered as ink! internals and not part of the library
+per-say.
+
+What this means is that any changes to the examples (breaking or not) should only be
+considered a `PATCH` level change. Additionally, they should not be published to
```
Comment (cmichi) [inline RELEASES_CHECKLIST.md:28]:
```suggestion
considered a `PATCH` level change. Additionally, they should not be published.
```

**Review by ascjones — APPROVED** [review]

File: RELEASES_CHECKLIST.md [hunk 3]
```diff
@@ -21,26 +18,51 @@ Reasons for doing it this way:
...
+crates.io.
+
+
 ## Checklist
 
-1. Make sure you've moved the changes in the `RELEASES.md` from `[Unreleased]` into a new section for the release.
-1. Check that all PRs since the last release are now in the new release section, in case the `[Unreleased]` section was incomplete.
+1. Create a new feature branch off `master`.
+1. Bump the version in all the TOML files to the new version.
```
Comment (cmichi) [inline RELEASES_CHECKLIST.md:35]:
```suggestion
1. Bump the version in all TOML files to the new version.
```

File: RELEASES_CHECKLIST.md [hunk 4]
```diff
@@ -21,26 +18,51 @@ Reasons for doing it this way:
...
+What this means is that any changes to the examples (breaking or not) should only be
+considered a `PATCH` level change. Additionally, they should not be published to
+crates.io.
+
+
 ## Checklist
 
-1. Make sure you've moved the changes in the `RELEASES.md` from `[Unreleased]` into a new section for the release.
-1. Check that all PRs since the last release are now in the new release section, in case the `[Unreleased]` section was incomplete.
+1. Create a new feature branch off `master`.
```
Comment (cmichi) [inline RELEASES_CHECKLIST.md:34]:
```suggestion
1. Create a new feature branch of `master`.
```

**Review by cmichi — APPROVED** [review]

**Pull request merged**